src/mprk.jl

```diff
@@ -703,40 +703,341 @@ function perform_step!(integrator, cache::MPRK22Cache, repeat_step = false)
+   discretizations of production–destruction systems."
+   BIT Numerical Mathematics 58 (2018): 691–728.
+  [DOI: 10.1007/s10543-018-0705-1](https://doi.org/10.1007/s10543-018-0705-1)
+"""
+struct MPRK43I{T, Thread, F} <: OrdinaryDiffEqAdaptiveAlgorithm
+    alpha::T
+    beta::T
+    thread::Thread
+    linsolve::F
+end
+
+function MPRK43I(alpha, beta; thread = False(), linsolve = LUFactorization())
+    MPRK43I{typeof(alpha), typeof(thread), typeof(linsolve)}(alpha, beta,
+                                                             thread, linsolve)
+end
+
+"""
+    MPRK43II(γ; [linsolve = ...])
+
+A third-order modified Patankar-Runge-Kutta algorithm for (conservative)
+production-destruction systems. This one-step, four-stage method is
+third-order accurate, unconditionally positivity-preserving, and linearly
+implicit. The parameters `γ` is described by Kopecz and Meister (2018).
+
+This modified Patankar-Runge-Kutta method requires the special structure of a
+[`PDSProblem`](@ref) or a [`ConservativePDSProblem`](@ref).
+
+You can optionally choose the linear solver to be used by passing an
+algorithm from [LinearSolve.jl](https://github.com/SciML/LinearSolve.jl)
+as keyword argument `linsolve`.
+
+## References
+
+- Stefan Kopecz and Andreas Meister.
+  "Unconditionally positive and conservative third order modified Patankar–Runge–Kutta 
+   discretizations of production–destruction systems."
+   BIT Numerical Mathematics 58 (2018): 691–728.
+  [DOI: 10.1007/s10543-018-0705-1](https://doi.org/10.1007/s10543-018-0705-1)
+"""
+struct MPRK43II{T, Thread, F} <: OrdinaryDiffEqAdaptiveAlgorithm
+    gamma::T
+    thread::Thread
+    linsolve::F
+end
+
+function MPRK43II(gamma; thread = False(), linsolve = LUFactorization())
+    MPRK43II{typeof(gamma), typeof(thread), typeof(linsolve)}(gamma, thread, linsolve)
+end
+
+# TODO: Consider switching to the interface of LinearSolve.jl directly,
+#       avoiding `dolinesolve` from OrdinaryDiffEq.jl.
+# TODO: Think about adding preconditioners to the algorithm
+# This hack is currently required to make OrdinaryDiffEq.jl happy...
+function Base.getproperty(alg::Union{MPRK43I, MPRK43II}, f::Symbol)
+    # preconditioners
+    if f === :precs
+        return Returns((nothing, nothing))
+    else
+        return getfield(alg, f)
+    end
+end
+
+alg_order(::MPRK43I) = 3
+alg_order(::MPRK43II) = 3
+isfsal(::MPRK43I) = false
+isfsal(::MPRK43II) = false
+
+struct MPRK43Cache{uType, rateType, PType, tabType, Thread, F, uNoUnitsType} <:
+       OrdinaryDiffEqMutableCache
+    u::uType
+    uprev::uType
+    tmp::uType
+    atmp::uType
+    k::rateType
+    fsalfirst::rateType
+    P::PType
+    P2::PType
+    D::uType
+    D2::uType
+    M::PType
+    σ::uType
+    tab::tabType
+    thread::Thread
+    linsolve_tmp::uType  # stores rhs of linear system
+    linsolve::F
+    weight::uNoUnitsType
+end
+
+function set_constant_parameters(alg::MPRK43I)
```

Comment:
```suggestion
function get_constant_parameters(alg::MPRK43I)
```
Just a minor thing I noticed while looking at the PR: To me a `set_*` function name sounds like a mutating function, but this rather looks like a `get_*` function to me.